pytorch_lightning/loops/dataloader/evaluation_dataloader_loop.py

```diff
@@ -0,0 +1,268 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, List, Optional, Sequence, Union
+
+from deprecate.utils import void
+from torch.utils.data.dataloader import DataLoader
+
+import pytorch_lightning as pl
+from pytorch_lightning.loops.dataloader.dataloader_loop import DataLoaderLoop
+from pytorch_lightning.loops.evaluation_epoch_loop import EvaluationEpochLoop
+from pytorch_lightning.trainer.connectors.logger_connector.result import ResultCollection
+from pytorch_lightning.trainer.states import TrainerFn
+from pytorch_lightning.utilities.model_helpers import is_overridden
+from pytorch_lightning.utilities.types import EPOCH_OUTPUT
+
+
+class EvaluationDataLoaderLoop(DataLoaderLoop):
+    """Loops over all dataloaders for evaluation."""
+
+    def __init__(self):
+        super().__init__()
+        self._max_batches: Optional[Union[int, Sequence[int]]] = None
+        self.outputs = []
+        self.evaluation_loop = EvaluationEpochLoop()
+
+        self._val_results = ResultCollection(training=False)
+        self._test_results = ResultCollection(training=False)
+
+    @property
+    def num_dataloaders(self) -> int:
+        """Returns the total number of dataloaders"""
+        # case where user does:
+        # return dl1, dl2
+        dataloaders = self.dataloaders
+        if dataloaders is None:
+            return 0
+        length = len(dataloaders)
+        if length > 0 and isinstance(dataloaders[0], (list, tuple)):
+            length = len(dataloaders[0])
+        return length
+
+    @property
+    def dataloaders(self) -> Sequence[DataLoader]:
+        """Returns the validation or test dataloaders"""
+        if self.trainer.testing:
+            return self.trainer.test_dataloaders
+        return self.trainer.val_dataloaders
+
+    @property
+    def results(self) -> Optional[ResultCollection]:
+        """Returns the current results"""
+        if self.trainer.validating or self.trainer.sanity_checking:
+            return self._val_results
+        elif self.trainer.testing:
+            return self._test_results
+        return None
+
+    @property
+    def predictions(self):
+        """Returns the predictions from all dataloaders"""
+        return self.evaluation_loop.predictions
+
+    def connect(self, trainer: "pl.Trainer", *args: Any, **kwargs: Any) -> None:
+        """Connects the loop to everything necessary (like trainer and accelerators)"""
+        super().connect(trainer, *args, **kwargs)
+        # TODO: Make the trainer a weakref/proxy
+        self.evaluation_loop.connect(trainer)
+
+    @property
+    def done(self) -> bool:
+        """Returns whether all dataloaders are processed or evaluation should be skipped altogether"""
+        return (self.current_dataloader_idx >= len(self.dataloaders)) or self.skip
+
+    @property
+    def skip(self) -> bool:
```

Comment:
```suggestion
    def should_skip(self) -> bool:
```
be more intuitive?

Reply:
can't change now, it is in the base loop and training loop too. 
Need to make a follow up if we want to change api for all loops :)